common/net/bolt1.py

```diff
@@ -0,0 +1,283 @@
+        if data_len:
+            data = message[36 : 36 + data_len]
+
+            if len(data) < data_len:
+                raise ValueError("Wrong message format. Unexpected EOF")
+            if len(message) > 36 + data_len:
+                raise ValueError("Wrong data format. message has additional tailing data")
+            return cls(channel_id, data.decode("utf-8"))
+
+        return cls(channel_id)
+
+
+class PingMessage(Message):
+    """
+    Message to test the reachability of the other side of the channel. Useful to allow long lived communications.
+
+    Args:
+        num_pong_bytes (:obj:`int`): the number of bytes to be responded by the peer.
+        ignored_bytes (:obj:`bytes`): filling bytes added to the message by the sender.
+    """
+
+    def __init__(self, num_pong_bytes, ignored_bytes=None):
+        if not 0 <= num_pong_bytes < pow(2, 16):
+            raise ValueError(f"num_pong_bytes must be between 0 and {pow(2, 16)}")
+
+        payload = num_pong_bytes.to_bytes(2, "big")
+
+        if ignored_bytes:
+            if not isinstance(ignored_bytes, bytes):
+                raise TypeError("ignored_bytes must be bytes if set")
+            if len(ignored_bytes) > pow(2, 16) - 4:
+                raise ValueError(f"ignored_bytes cannot be higher than {pow(2, 16) - 4}")
+            payload += len(ignored_bytes).to_bytes(2, "big") + ignored_bytes
+
+        super().__init__(message_types["ping"], payload)
+        self.num_pong_bytes = num_pong_bytes
+        self.ignored_bytes = ignored_bytes
+
+    @classmethod
+    def from_bytes(cls, message):
+        """Builds a PingMessage from its byte representation."""
+        # Message should be at least: type (2-byte) + num_pong_bytes (2-bytes) + byteslen (2-bytes)
+        message_sanity_checks(message, message_types["ping"], 6)
+        num_pong_bytes = int.from_bytes(message[2:4], "big")
+        byteslen = int.from_bytes(message[4:6], "big")
+
+        if byteslen:
+            ignored = message[6 : 6 + byteslen]
+            if len(ignored) < byteslen:
+                raise ValueError("Wrong message format. Unexpected EOF")
+            if len(message) > 6 + byteslen:
+                raise ValueError("Wrong data format. message has additional tailing data")
+            return cls(num_pong_bytes, ignored)
+
+        return cls(num_pong_bytes)
+
+
+class PongMessage(Message):
+    """
+    Message to be sent in response to a ``PingMessage``.
+
+    Args:
+        ignored_bytes (:obj:`bytes`): filling bytes added to the message by the sender. Should match the ones requested
+            by the sender of the ``PingMessage``.
+    """
+
+    def __init__(self, ignored_bytes=None):
+        if ignored_bytes:
+            if not isinstance(ignored_bytes, bytes):
+                raise TypeError("ignored_bytes must be bytes if set")
+            if len(ignored_bytes) > pow(2, 16) - 4:
+                raise ValueError(f"ignored_bytes cannot be higher than {pow(2, 16) -4}")
```

Comment:
missing space `-4` => `- 4`.

Reply:
Fixed in d744359630eae32899372242661994ca1ef294d7